react-basic-practice/src/contexts/loading.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { useContext, useMemo, useState } from 'react';
+
+const LoadingContext = React.createContext();
+
+const useLoading = () => useContext(LoadingContext);
+
+const LoadingProvider = (props) => {
+  const { children } = props;
+  const [loading, setLoading] = useState(false);
+  const show = () => setLoading(true);
+  const hide = () => setLoading(false);
+  const valueContext = useMemo(() => ({
```

Comment:
Why useMemo?

Reply:
As the non-stable values should not be used as the value for `Context.Provider` so I use `useMemo` to wrap the value.
[This rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-constructed-context-values.md).